### PR TITLE
fix(HII-5773): timeout error despite topic publishing messages

### PR DESCRIPTION
**Pull request by eetrip**

### CODE REVIEW

File: src/pubsub-publisher/pubsub-publisher.service.ts [hunk 1]
```diff
@@ -37,36 +37,36 @@ export class PubSubPublisherService {
...
+          totalTimeoutMillis: this.settings.requestTimeoutMillis,
+        },
+      },
+    });
 
-    timeout.clear();
+    const [id] = messageIds || [''];
 
-    return result;
+    return id;
```
Comment (glebbash) [inline src/pubsub-publisher/pubsub-publisher.service.ts]: `return messageIds[0] ?? '';` would be better

File: src/pubsub-publisher/pubsub-publisher.service.ts [hunk 2]
```diff
@@ -37,36 +37,36 @@ export class PubSubPublisherService {
     data: Buffer,
     attributes: Record<string, string>
   ): Promise<string> {
-    const timeout = setTimeoutAsync(this.settings.requestTimeoutMillis);
+    const publisherClient = new v1.PublisherClient({});
+    const formattedTopic = publisherClient.projectTopicPath(projectId, topic.name);
+    const dataBuffer = Buffer.from(data);
```
Comment (glebbash) [inline src/pubsub-publisher/pubsub-publisher.service.ts]: Are you sure this is using the same pubsub config as the previous solution?

I think it creates a new PubSub instance in `new v1.PublisherClient({});`.

Also there is probably a way to get PublisherClient from `PubSub` or `Topic` instances.

**Review by glebbash — CHANGES_REQUESTED** [review]
We should not have JIRA ticket names in the public repositories, please change PR title and branch name. Also see other comments.

File: src/pubsub-publisher/pubsub-publisher.service.ts (hunk 1 above)
Comment (eetrip) [reply src/pubsub-publisher/pubsub-publisher.service.ts]: agreed

File: src/pubsub-publisher/pubsub-publisher.service.ts (hunk 2 above)
Comment (eetrip) [reply src/pubsub-publisher/pubsub-publisher.service.ts]: there might be a way to get publisher client from topic instance, was trying to apply that while fixing tests, will verify

**Pull request closed without merging**